papaya-core/src/main/java/pl/edu/agh/papaya/security/UserService.java

```diff
@@ -0,0 +1,51 @@
+package pl.edu.agh.papaya.security;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import javax.ws.rs.NotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.keycloak.admin.client.resource.UsersResource;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final KeycloakProvider keycloakProvider;
+
+    public Optional<User> getUserById(String userId) {
+        return getUserBy(resource -> resource.get(userId).toRepresentation());
+    }
+
+    private Optional<User> getUserBy(Function<UsersResource, UserRepresentation> selector) {
```

Comment:
średnio podoba mi się, że pobieramy wszystkich userów i gdzieś dopiero tutaj filtrujemy do tych co chcemy, ale nie wiem jak to z keycloakiem się robi, więc tak tylko zwracam uwagę

Reply:
Metoda `users` nam nie zwraca wszystkich userów tylko zasób związany z userami. Na tym zasobie jak wywołany konkretna metodę to wtedy dopiero leci request.

Jedyne co mi się średnio podoba jest to że w notification service pytamy o każdego usera z osobna. Na razie zrobiłem tak bo nie znalazłem innego rozwiązania, jak się okaże że nie ma to wtedy zrobimy cache.